config/opal_config_threads.m4

```diff
@@ -34,36 +35,74 @@ AC_DEFUN([OPAL_CONFIG_THREADS],[
-AC_MSG_CHECKING([for working POSIX threads package])
-if test "$HAVE_POSIX_THREADS" = "1" ; then
-  AC_MSG_RESULT([yes])
-else
-  AC_MSG_RESULT([no])
-fi
-export HAVE_POSIX_THREADS
+
+AC_ARG_WITH([threads],
+            [AC_HELP_STRING([--with-threads=TYPE],
+                        [Specify thread TYPE to use. default:pthreads. Other options are qthreads and argobots.])])
+
+#
+# Check we for the thread package requested, or posix
+#
+
+thread_type_found=
+
+#
+# check for posix threads
+#
+AS_IF([test -z "$with_threads" || test "$with_threads" = "pthreads" || test "$with_threads" = "yes"],
+      [OPAL_CONFIG_POSIX_THREADS(HAVE_THREAD_PKG=1, HAVE_THREAD_PKG=0)
+       AC_MSG_CHECKING([for working POSIX threads package])
+       AS_IF([test "$HAVE_THREAD_PKG" = "1"],
+             [AC_MSG_RESULT([yes])
+              thread_type_found="pthreads"],
+             [AC_MSG_RESULT([no])])],
+      [])
+
+#
+# see if argobots is called for
+#
+AS_IF([test -z "$thread_type_found" && test "$with_threads" = "argobots"],
+      [OPAL_CONFIG_ARGOBOTS_THREADS(HAVE_THREAD_PKG=1, HAVE_THREAD_PKG=0)
+       AC_MSG_CHECKING([for working ARGOBOTS threads package])
+       AS_IF([test "$HAVE_THREAD_PKG" = "1"],
+             [AC_MSG_RESULT([yes])
+              thread_type_found="argobots"],
+             [AC_MSG_RESULT([no])])],
+      [])
+
+AS_IF([test -z "$thread_type_found" && test "$with_threads" = "qthreads"],
+      [OPAL_CONFIG_QTHREADS(HAVE_THREAD_PKG=1, HAVE_THREAD_PKG=0)
+       AC_MSG_CHECKING([for working Qthreads package])
+       AS_IF([test "$HAVE_THREAD_PKG" = "1"],
+             [AC_MSG_RESULT([yes])
+              thread_type_found="qthreads"],
+             [AC_MSG_RESULT([no])])],
+      [])
 
 #
-# Ask what threading we want (allow posix right now)
+# Bail if we didn't find any thread package
 #
 
-if test "$HAVE_POSIX_THREADS" = "0"; then
-    AC_MSG_WARN(["*** POSIX threads are not"])
-    AC_MSG_WARN(["*** available on your system "])
-    AC_MSG_ERROR(["*** Can not continue"])
-fi
+AS_IF([test -z "$thread_type_found"],
+      [AC_MSG_WARN([*** no thread package $with_threads])
+       AC_MSG_WARN([*** available on your system])
+       AC_MSG_ERROR([*** Can not continue])])
 
-THREAD_CFLAGS="$PTHREAD_CFLAGS"
-THREAD_FCFLAGS="$PTHREAD_FCFLAGS"
-THREAD_CXXFLAGS="$PTHREAD_CXXFLAGS"
-THREAD_CPPFLAGS="$PTHREAD_CPPFLAGS"
-THREAD_CXXCPPFLAGS="$PTHREAD_CXXCPPFLAGS"
-THREAD_LDFLAGS="$PTHREAD_LDFLAGS"
-THREAD_LIBS="$PTHREAD_LIBS"
+THREAD_CFLAGS="$TPKG_CFLAGS"
+THREAD_FCFLAGS="$TPKG_FCFLAGS"
+THREAD_CXXFLAGS="$TPKG_CXXFLAGS"
+THREAD_CPPFLAGS="$TPKG_CPPFLAGS"
+THREAD_CXXCPPFLAGS="$TPKG_CXXCPPFLAGS"
+THREAD_LDFLAGS="$TPKG_LDFLAGS"
+THREAD_LIBS="$TPKG_LIBS"
+HAVE_THREAD_PKG_TYPE="$thread_type_found"
+export HAVE_THREAD_PKG_TYPE
 
-OPAL_CHECK_PTHREAD_PIDS
+AS_IF([test "$thread_type_found" = "pthreads"],
+      [OPAL_CHECK_PTHREAD_PIDS],[])
 
+OPAL_SUMMARY_ADD([[Miscellaneous]],[[Threading Package]],[opal_threads], [$thread_type_found])
```

Comment:
Now that threads is a framework, shouldn't all of this be in `opal/mca/threads/configure.m4` and/or the various component `configure.m4` files?  Seems like an abstraction violation to have all this stuff at the top level now that there is a framework and components for each of these individual things.

Reply:
done